katello-attach-subscription

```diff
@@ -201,53 +918,177 @@ def vdcupdate()
+                puts "    Starting removing subscriptions from Hypervisor"
+                rem_sub_response["results"].each_with_index do |rem_sub, currentsub|
+                  if @options[:verbose]
+                    puts "      VERBOSE: Removing subscription #{currentsub+1} / #{rem_sub_response["total"].to_i} - #{rem_sub['id']}"
+                  end
+                  if @options[:debug]
+                    puts "      DEBUG: Current subscription data"
+                    p rem_sub
+                  end
+                  begin
+                    # removing the subscription
+                    if not @options[:noop]
+                      @api.resource(:host_subscriptions).call(:remove_subscriptions, {:host_id => sys['id'], :subscriptions => [{:id => rem_sub['id']}]})
+                      puts "      Removed subscription #{rem_sub['id']} from hypervisor #{sys['name']} / #{sys['id']}"
+                    else
+                      puts "      [noop]: sub #{rem_sub['id']} would be removed from hypervisor #{sys['name']} / #{sys['id']}"
+                    end
+                  rescue Exception => e
+                    puts "      ERROR: Unknow Error - Unable to remove subscription #{rem_sub["id"]} from #{sys["name"]}"
+                    puts e.message
+                    puts e.backtrace.inspect
+                    puts e.response
+                    exit 1
+                  end
+                end
+              else
+                puts "    No subscription found attached to #{sys['name']}. Nothing to do."
+              end
+            else
+              puts "  Hypervisor #{sys['name']} is in #{this_system_cluster} that is a full cluster. Proced to subcription attaching"
+            end
           end
-          if virtual_host_name
-            if not virtualhostregex.match(virtual_host_name)
-              puts "Skipping '#{system['name']}' as the system should be on #{sub['virtual_host']} but it is on #{virtual_host_name}"
-              next
+
+          if @options[:emptyhypervisor]
+            # check if virtual_guests array in subscription_facet_attributes has elements, if empty skip to the next hypervisor
+            if @options[:verbose]
+              puts "    VERBOSE: Checking if #{sys['name']} has virtual_guests on it."
+            end
+            if sys["subscription_facet_attributes"].has_key?("virtual_guests") and sys["subscription_facet_attributes"]["virtual_guests"].count > 0
+              if @options[:verbose]
+                puts "      VERBOSE: Hosts isn't empty, proceed to subscription"
+              end
+              if @options[:debug]
+                puts "        DEBUG: Number of guests: #{sys["subscription_facet_attributes"]["virtual_guests"].count}"
+              end
+            else
+              if @options[:verbose]
+                puts "      VERBOSE: Skipping as #{sys["name"]} has no guests in it"
+              end
+              skip_hypervisor = true
```

Comment:
`skip_hypervisor ||= hypervisor_has_guests(sys)`

and `hypervisor_has_guests` is a function that does all the above